### PR TITLE
BackgroundTransition fixes

### PR DESCRIPTION
**Pull request by ramezgerges**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/BorderHelper.skia.cs [hunk 1]
```diff
@@ -47,31 +47,42 @@ public static void UpdateAllBorderProperties(this IBorderInfoProvider @this)
...
 
-		var oldBrush = fromBrush as SolidColorBrush;
-		var newBrush = toBrush as SolidColorBrush;
+		if (isAnimation)
+		{
+			// When an animation sets the background, it skips the transition logic and is applied immediately.
+			// However, it "deactivates" the currently active transition if one exists. Once the animation is done
+			// the deactivated transition is reactivated and continues as if it was running (i.e. it's not paused,
+			// but takes into account the duration during which it was deactivated).
+			visual.Compositor.DeactivateBackgroundTransition(visual);
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/BorderHelper.skia.cs:63]: Could we add a `return;` after this line and remove the else block?


Also, this method was a port from WinUI, so unclear why we need an Uno-specific workaround

File: src/Uno.UI/UI/Xaml/Controls/Border/Border.cs [hunk 2]
```diff
@@ -311,7 +311,12 @@ protected override void OnBackgroundChanged(DependencyPropertyChangedEventArgs e
 	{
 #if UNO_HAS_BORDER_VISUAL
 		this.UpdateBackground();
-		BorderHelper.SetUpBrushTransitionIfAllowed((BorderVisual)this.Visual, e.OldValue as Brush, e.NewValue as Brush, this.BackgroundTransition);
+		BorderHelper.SetUpBrushTransitionIfAllowed(
+			(BorderVisual)this.Visual,
+			e.OldValue as Brush,
+			e.NewValue as Brush,
+			this.BackgroundTransition,
+			(this as IDependencyObjectStoreProvider).Store.GetPropertyDetails(BackgroundProperty).CurrentHighestValuePrecedence == DependencyPropertyValuePrecedences.Animations);
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/Border/Border.cs:319]: This should be a direct cast

File: src/Uno.UI/UI/Xaml/Controls/ContentPresenter/ContentPresenter.cs [hunk 3]
```diff
@@ -1219,7 +1219,12 @@ protected override void OnBackgroundChanged(DependencyPropertyChangedEventArgs e
 	{
 #if UNO_HAS_BORDER_VISUAL
 		this.UpdateBackground();
-		BorderHelper.SetUpBrushTransitionIfAllowed((BorderVisual)this.Visual, e.OldValue as Brush, e.NewValue as Brush, this.BackgroundTransition);
+		BorderHelper.SetUpBrushTransitionIfAllowed(
+			(BorderVisual)this.Visual,
+			e.OldValue as Brush,
+			e.NewValue as Brush,
+			this.BackgroundTransition,
+			(this as IDependencyObjectStoreProvider).Store.GetPropertyDetails(BackgroundProperty).CurrentHighestValuePrecedence == DependencyPropertyValuePrecedences.Animations);
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/ContentPresenter/ContentPresenter.cs:1227]: Same here

File: src/Uno.UI/UI/Xaml/Controls/Panel/Panel.cs [hunk 4]
```diff
@@ -239,7 +239,12 @@ protected override void OnBackgroundChanged(DependencyPropertyChangedEventArgs e
 	{
 #if UNO_HAS_BORDER_VISUAL
 		this.UpdateBackground();
-		BorderHelper.SetUpBrushTransitionIfAllowed((BorderVisual)this.Visual, e.OldValue as Brush, e.NewValue as Brush, this.BackgroundTransition);
+		BorderHelper.SetUpBrushTransitionIfAllowed(
+			(BorderVisual)this.Visual,
+			e.OldValue as Brush,
+			e.NewValue as Brush,
+			this.BackgroundTransition,
+			(this as IDependencyObjectStoreProvider).Store.GetPropertyDetails(BackgroundProperty).CurrentHighestValuePrecedence == DependencyPropertyValuePrecedences.Animations);
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/Panel/Panel.cs:247]: Same here

File: src/Uno.UI/UI/Xaml/BorderHelper.skia.cs [hunk 5]
```diff
@@ -47,13 +47,23 @@ public static void UpdateAllBorderProperties(this IBorderInfoProvider @this)
...
 
-	public static void SetUpBrushTransitionIfAllowed(BorderVisual visual, Brush fromBrush, Brush toBrush, BrushTransition transition)
+	public static void SetUpBrushTransitionIfAllowed(BorderVisual visual, Brush fromBrush, Brush toBrush, BrushTransition transition, bool isAnimation)
 	{
 		if (transition is null)
 		{
 			return;
 		}
 
+		if (isAnimation)
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/BorderHelper.skia.cs:57]: As this method is ported code, best to mark with Uno-specific comment. Still unclear to me why we need Uno-specific code though. What exactly is missing to match WinUI?

**Review by Youssef1313 — APPROVED** [review]
LGTM with small comment regarding Uno-specific logic. Also best if you could add a test

File: src/Uno.UI/UI/Xaml/BorderHelper.skia.cs (hunk 5 above)
Comment (ramezgerges) [reply src/Uno.UI/UI/Xaml/BorderHelper.skia.cs:57]: What the comment says:
```
// When an animation sets the background, it skips the transition logic and is applied immediately.
// However, it "deactivates" the currently active transition if one exists. Once the animation is done,
// the deactivated transition is reactivated and continues as if it was running (i.e. it's not paused,
// but takes into account the duration during which it was deactivated).
```

**Pull request merged**